iis-wf/iis-wf-import/src/test/java/eu/dnetlib/iis/wf/importer/infospace/truncator/DocumentMetadataAvroTruncatorTest.java

```diff
@@ -0,0 +1,156 @@
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DocumentMetadataAvroTruncatorTest {
+
+    @Mock
+    private Function<CharSequence, CharSequence> abstractTruncator;
+
+    @Mock
+    private Function<CharSequence, CharSequence> titleTruncator;
+
+    @Mock
+    private Function<List<Author>, List<Author>> authorsTruncator;
+
+    @Mock
+    private Function<List<CharSequence>, List<CharSequence>> keywordsTruncator;
+
+    @InjectMocks
+    private DocumentMetadataAvroTruncator documentMetadataAvroTruncator = DocumentMetadataAvroTruncator.newBuilder().build();
+
+    @Test
+    public void shouldNotTruncateAbstractWhenItIsNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals(source, result);
+        verify(abstractTruncator, never()).apply(any());
+    }
+
+    @Test
+    public void shouldTruncateAbstractWhenItIsNotNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+        source.setAbstract$("abstract");
+        when(abstractTruncator.apply("abstract")).thenReturn("truncated");
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals("truncated", result.getAbstract$());
+    }
+
+    @Test
+    public void shouldNotTruncateTitleWhenItIsNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals(source, result);
+        verify(titleTruncator, never()).apply(any());
+    }
+
+    @Test
+    public void shouldTruncateTitleWhenItIsNotNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+        source.setTitle("title");
+        when(titleTruncator.apply("title")).thenReturn("truncated");
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals("truncated", result.getTitle());
+    }
+
+    @Test
+    public void shouldNotTruncateAuthorsWhenItIsNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals(source, result);
+        verify(authorsTruncator, never()).apply(any());
+    }
+
+    @Test
+    public void shouldTruncateAuthorsWhenItIsNotNull() {
```

Comment:
I guess we could also add a test verifying whether the number of authors is properly truncated.

Reply:
This unit test checks if authors truncation logic is triggered. It uses a stubbed mock for `Function<List<Author>, List<Author>>` that is responsible for truncation of authors field in `DocumentMetadataAvroTruncator`. So this test method simply checks if the result of truncation is the same as the value used for stubbing. The actual size of the list is not important, that's why I've used a `Collections.singletonList(mock(Author.class))`. The 'real' truncation mechanism is checked in `ImportInformationSpaceJobTest` and in workflow integration test.

Reply:
OK then. I thought we are missing the unit test proving maxAuthorsSize limitation works as expected.